nomenclature/config.py

```diff
@@ -159,9 +159,15 @@ class NomenclatureConfig(BaseModel):
     repositories: dict[str, Repository] = Field(default_factory=dict)
     definitions: DataStructureConfig = Field(default_factory=DataStructureConfig)
     mappings: RegionMappingConfig = Field(default_factory=RegionMappingConfig)
+    illegal_characters: None | list[str] = None
```

Comment:
One more meta comment on the default value of iterable attributes (lists, dicts and sets). Recently, I've liked using an empty version instead of `None`. Reason being, that it's nicer to handle downstream since it saves you from implementing a bunch of checks if an attribute is None.
The whole config class is written around that idea which makes the code easier to read and write.
As an example, above you have the following check

```python
        if config and config.illegal_characters:
            forbidden += config.illegal_characters
```

if you used an empty list as a default you don't need the second check in the if:
```python
        if config and config.illegal_characters:
            forbidden += config.illegal_characters
```
since worst case you're adding an empty list.

As I said, the nomenclature config class is written around this idea of avoiding `None` as a default. This means that as a default for the config in the `check_illegal_characters` function, you can (and should in my opinion) use `NomenclatureConfig`.

This way your code can be further shortened to:

```python
    forbidden += config.illegal_characters
```
and you're actually saving yourself a level of nesting, making the code even nicer to read. 
If you look at how check_illegal_characters is called you can see that self.config can actually never be `None` so you don't even need a default value.

The reason that I use the `default_factory` option with pydantic has to do with the caveats of using mutable default values.

Reply:
Opening this up again. @dc-almeida, it seems that you marked this conversation as resolved without implementing the suggestion or saying why you won't implement it.
Please either implement the suggestion or provide a reason for not implementing.

Reply:
Apologies, yesterday I did the commit but hadn't been pushed. Should be good now.